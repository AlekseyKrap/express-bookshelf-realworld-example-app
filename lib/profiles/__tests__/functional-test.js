'use strict';

const chance = require('chance').Chance('profile');
const request = require('supertest');
const app = require('../../app');

const {users} = app.locals.services;

describe('profiles', () => {
  let userOne;
  let userTwo;

  beforeEach(async () => {
    userOne = await users.create({
      bio: chance.profession(),
      email: chance.email(),
      image: chance.avatar(),
      password: chance.word({length: 8}),
      username: chance.word(),
    });
    userTwo = await users.create({
      bio: chance.profession(),
      email: chance.email(),
      image: chance.avatar(),
      password: chance.word({length: 8}),
      username: chance.word(),
    });
  });

  afterEach(async () => {
    await userOne.destroy();
    await userTwo.destroy();
  });

  describe('GET /api/profiles/:username', () => {
    describe('with invalid username', () => {
      test('should return a 404', async () => {
        const {body, status} = await request(app)
          .get('/api/profiles/invalid')
          .set('Accept', 'application/json')
          .send();

        expect({body, status}).toMatchSnapshot();
      });
    });

    describe('without logged in user', () => {
      test('should return a 200', async () => {
        const {body, status} = await request(app)
          .get(`/api/profiles/${userOne.get('username')}`)
          .set('Accept', 'application/json')
          .send();

        expect({body, status}).toMatchSnapshot();
      });
    });

    describe('with logged in user', () => {
      test('should return a 200', async () => {
        const {body, status} = await request(app)
          .get(`/api/profiles/${userTwo.get('username')}`)
          .set('Accept', 'application/json')
          .set('Authorization', `Token ${users.generateJWT(userOne)}`)
          .send();

        expect({body, status}).toMatchSnapshot();
      });
    });
  });
});
