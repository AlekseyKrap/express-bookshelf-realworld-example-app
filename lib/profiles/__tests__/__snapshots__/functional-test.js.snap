// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profiles GET /api/profiles/:username with invalid username should return a 404 1`] = `
Object {
  "body": Object {
    "errors": Object {
      "error": Object {
        "data": null,
        "isBoom": true,
        "isServer": false,
        "message": "Not Found",
        "output": Object {
          "headers": Object {},
          "payload": Object {
            "error": "Not Found",
            "message": "Not Found",
            "statusCode": 404,
          },
          "statusCode": 404,
        },
      },
      "message": "Not Found",
    },
  },
  "status": 404,
}
`;

exports[`profiles GET /api/profiles/:username with logged in user should return a 200 1`] = `
Object {
  "body": Object {
    "profile": Object {
      "bio": "Contract Administrator",
      "following": false,
      "image": "//www.gravatar.com/avatar/381695debd1a3738c5b4bb61d125ceb9",
      "username": "wochaco",
    },
  },
  "status": 200,
}
`;

exports[`profiles GET /api/profiles/:username without logged in user should return a 200 1`] = `
Object {
  "body": Object {
    "profile": Object {
      "bio": "Interpreter",
      "following": false,
      "image": "//www.gravatar.com/avatar/95a4206b9d5f87c464fd32e7b3a67ef7",
      "username": "ela",
    },
  },
  "status": 200,
}
`;
